### PR TITLE
Issue #557 Make other nested query options in  to work

### PR DESCRIPTION
**Pull request by VikingsFan**

### CODE REVIEW

File: OData/src/System.Web.OData/OData/Query/Expressions/SelectExpandBinder.cs [hunk 1]
```diff
@@ -461,8 +461,19 @@ private Expression CreateTotalCountExpression(Expression source, ExpandedNavigat
...
+                LambdaExpression orderByExpression =
+                    FilterBinder.Bind(orderbyClause, elementType, _model, _settings);
+                source = ExpressionHelpers.OrderBy(source, orderByExpression, elementType, orderbyClause.Direction);
+            }
+            return source;
+        }
+
         // new CollectionWrapper<ElementType> { Instance = source.Select((ElementType element) => new Wrapper { }) }
-        private Expression ProjectCollection(Expression source, Type elementType, SelectExpandClause selectExpandClause, IEdmEntityType entityType, IEdmEntitySet entitySet)
+        private Expression ProjectCollection(ExpandedNavigationSelectItem expandedItem, Expression source, Type elementType, SelectExpandClause selectExpandClause, IEdmEntityType entityType, IEdmEntitySet entitySet)
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Query/Expressions/SelectExpandBinder.cs]: why do you put the new added parameter at head, while other function at tail?

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ private static IEdmModel GetEdmModel(HttpConfiguration configuration)
             EntitySetConfiguration<SelectCustomer> customers = builder.EntitySet<SelectCustomer>("SelectCustomer");
             customers.EntityType.Action("CreditRating").Returns<double>();
             customers.EntityType.Collection.Action("PremiumCustomers").ReturnsCollectionFromEntitySet<SelectCustomer>("SelectCustomer");
-
+
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: revert

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs [hunk 3]
```diff
@@ -510,6 +510,122 @@ public void NestedNestedDollarCountInDollarExpandWorks()
...
+        public void NestedNestedSkipInDollarExpandWorks()
+        {
+            string queryUrl = string.Format("{0}/selectexpand/SelectCustomer?" +
+                    "$expand=SelectOrders($skip=1;$expand=OrderDetails($skip=1))", BaseAddress);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, queryUrl);
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json;odata.metadata=none"));
+            HttpClient client = new HttpClient();
+            HttpResponseMessage response = client.SendAsync(request).Result;
+
+            // Assert
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: only assert?

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs [hunk 4]
```diff
@@ -510,6 +510,122 @@ public void NestedNestedDollarCountInDollarExpandWorks()
...
+        [Fact]
+        public void NestedNestedOrderByInDollarExpandWorks()
+        {
+            string queryUrl = string.Format("{0}/selectexpand/SelectCustomer?" +
+                    "$expand=SelectOrders($orderby=BillingAddress/ZipCode;$expand=OrderDetails($orderby=Price desc))", BaseAddress);
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, queryUrl);
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json;odata.metadata=none"));
+            HttpClient client = new HttpClient();
+            HttpResponseMessage response = client.SendAsync(request).Result;
+
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: remove the WS

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs [hunk 5]
```diff
@@ -537,7 +653,7 @@ public SelectCustomerController()
                     {
                         FirstLine = string.Format("First line {0}", j),
                         SecondLine = string.Format("Second line {0}", j),
-                        ZipCode = j * 100,
+                        ZipCode = j * -100,
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: why change this?

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs [hunk 6]
```diff
@@ -765,7 +883,7 @@ public class SelectOrder
         public int Id { get; set; }
         public DateTimeOffset Date { get; set; }
         public SelectAddress BillingAddress { get; set; }
-        public virtual IList<SelectOrderDetail> OrderDetails { get; set; }
+        public virtual IQueryable<SelectOrderDetail> OrderDetails { get; set; }
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: how about the test for IEnumerable?

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs (hunk 5 above)
Comment (VikingsFan) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: to make specific orderby result

File: OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs (hunk 6 above)
Comment (VikingsFan) [inline OData/test/E2ETest/WebStack.QA.Test.OData/QueryComposition/SelectExpandTests.cs]: SelectOrder is IEnumerable, this change is for Test both IEnumerable and IQueryable

**Comment by VikingsFan** [comment]
Merged 3f568901da0a4d3b7d7aaecb29f66e5ce7e14429

**Pull request closed without merging**